frontend/sass/_header.scss

```diff
@@ -20,6 +20,7 @@
   border-bottom: none;
   min-height: 4rem;
   padding: $spacing-thin 0 $spacing-thin 0;
+  z-index: auto;
```

Comment:
This fixes the mobile nav issue. `z-index` was being applied on a non-positioned element, causing issues with the nested elements.